src/iface/fragmentation.rs

```diff
@@ -3,28 +3,28 @@
 use managed::{ManagedMap, ManagedSlice};
 
 use crate::storage::Assembler;
-use crate::time::Instant;
+use crate::time::{Duration, Instant};
 use crate::Error;
 use crate::Result;
 
 /// Holds different fragments of one packet, used for assembling fragmented packets.
 #[derive(Debug)]
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct PacketAssembler<'a> {
     buffer: ManagedSlice<'a, u8>,
     assembler: AssemblerState,
 }
 
 /// Holds the state of the assembling of one packet.
 #[derive(Debug, PartialEq)]
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 enum AssemblerState {
     NotInit,
+    Discarded,
     Assembling {
         assembler: Assembler,
-        total_size: usize,
+        total_size: Option<usize>,
         last_updated: Instant,
         started_on: Instant,
+        ttl: Duration,
```

Comment:
I think storing just one `expires_at: Instant` would simplify the logic a lot.

Reply:
Yes indeed! Will change it.